include/wil/com_apartment_variable.h

```diff
@@ -169,6 +170,50 @@ namespace details
         // Apartment id -> variables storage.
         inline static wil::object_without_destructor_on_shutdown<std::unordered_map<unsigned long long, apartment_variable_storage>> s_apartmentStorage;
 
+        // NOTE: Requires 's_lock' to be held
+        static apartment_variable_storage* get_current_apartment_variable_storage()
+        {
+            auto storage = s_apartmentStorage.get().find(test_hook::GetApartmentId());
+            if (storage != s_apartmentStorage.get().end())
+            {
+                return &storage->second;
+            }
+            return nullptr;
+        }
+
+        // NOTE: Requires 's_lock' to be held
+        apartment_variable_storage* ensure_current_apartment_variables()

```

Comment:
If it requires the lock to be held should it take one of those WIL lock holder parameters to try and enforce that?